dspace-spring-rest/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -148,18 +173,115 @@ public ResourceSupport findRel(HttpServletRequest request, @PathVariable String
 		return findRelInternal(request, apiCategory, model, uuid, rel, page, assembler, projection);
 	}
 
-	@RequestMapping(method = RequestMethod.GET, value = "/{id:[A-z0-9]+}/{rel}/{relid:[A-z0-9]+}")
+	@RequestMapping(method = RequestMethod.GET, value = "/{id:^(?!^\\d+$)(?!^[0-9a-fxA-FX]{8}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{12}$)[\\w+\\-]+$}/{rel}/{relid:[\\w+\\-]+}")
```

Comment:
Please add a comment explaining this very large regex. It is not self-explanatory.